### ISSUE TITLE
DatetimePicker value not updated in other object's callbacks

### ISSUE THREAD

**Issue by thuydotm** [holoviz/panel#3993]
#### Description of expected behavior and the observed behavior
Let's say we have a DatetimePicker, and a button "Click". We change the date time value, then directly click on the button. In the callback of the button, the selected value is not updated properly. It still shows the time before the latest selection.

It works OK if I first change the date time, click outside to close the pop up, and finally click on the button. 

#### Complete, minimal, self-contained example code that reproduces the issue

```
import panel as pn

dt = pn.widgets.DatetimePicker()

def watcher(event):
    print(event.new, dt.value)

btn = pn.widgets.Button(name='Click')

dt_watcher = dt.param.watch(watcher, 'value', onlychanged=True)
btn.param.watch(watcher, 'value', onlychanged=True)

pn.Row(dt, btn).servable()
```

**Comment by Hoxbro** [comment]
I will fix this in #3990

**Comment by Hoxbro** [comment]
After reading through your description a second time. I think the problem comes down to timing. The DatetimePicker updates on the close of the pop-up calendar, and when you click directly on the button, it doesn't have time to sync with the python side. 

Nonetheless, I think #3990 will improve the experience.

**Comment by philippjfr** [comment]
Yeah, this is an unfortunate consequence of the fact that events (like a button click) are handled immediately while model change events (like the value change of the `DatetimePicker`) are slightly throttled. Not sure I have a good suggestion for a fix.
